src/replication.c

```diff
@@ -1767,7 +1773,8 @@ static int useDisklessLoad(void) {
             enabled = 0;
         }
         /* Check all modules handle async replication, otherwise it's not safe to use diskless load. */
-        else if (server.repl_diskless_load == REPL_DISKLESS_LOAD_SWAPDB && !moduleAllModulesHandleReplAsyncLoad()) {
+        else if ((server.repl_diskless_load == REPL_DISKLESS_LOAD_SWAPDB || shouldFallbackToDisklessLoad()) 
```

Comment:
Not sure how that check is needed? The disklessload fallback will not preserve any data right? so why do we want to support async replication in such case?